src/common/sharedlibrary/CMakeLists.txt

```diff
@@ -14,6 +14,24 @@
 #
 # Authored by: Alexandros Frantzis <alexandros.frantzis@canonical.com>
 
+include(CheckCXXSymbolExists)
+
+list(APPEND CMAKE_REQUIRED_LIBRARIES dl)
+list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
```

Comment:
With `check_cxx_symbol_exists()` these are not needed.

Reply:
This does not appear to be correct (though removing them was a good test of my "Error if symbol detextion isn't working" commit.

Reply:
Removing `list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)` doesn't seem to cause a problem, but it also doesn't break the test before the switch to check_cxx_symbol_exists(), so I don't want to take it out unless I know why RAOF put it there before.

Reply:
`list(APPEND CMAKE_REQUIRED_LIBRARIES dl)` was needed by `check_function_exists()`
`list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)` was always unnecessary.

Reply:
The latter may be correct, the former appears not to be. Try taking it out and see both `dlvsym` and `dlsym` stop getting detected by `check_cxx_symbol_exists()`